quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/web/QuarkusRequestFilter.java

```diff
@@ -1,175 +0,0 @@
-import io.vertx.core.http.HttpServerRequest;
-import io.vertx.ext.web.RoutingContext;
-
-/**
- * <p>This filter is responsible for managing the request lifecycle as well as setting up the necessary context to process incoming
- * requests. We need this filter running on the top of the chain in order to push contextual objects before executing Resteasy. It is not
- * possible to use a {@link jakarta.ws.rs.container.ContainerRequestFilter} for this purpose because some mechanisms like error handling
- * will not be able to access these contextual objects.
- * 
- * <p>The filter itself runs in an event loop and should delegate to worker threads any blocking code (for now, all requests are handled
- * as blocking).
- *
- * <p>Note that this filter is only responsible to close the {@link KeycloakSession} if not already closed when running Resteasy code. The reason is that closing it should be done at the
- * Resteasy level so that we don't block event loop threads even if they execute in a worker thread. Vert.x handlers and their
- * callbacks are not designed to run blocking code. If the session is eventually closed here is because Resteasy was not executed.
- *
- * @see org.keycloak.quarkus.runtime.integration.jaxrs.TransactionalResponseInterceptor
- * @see org.keycloak.quarkus.runtime.integration.jaxrs.TransactionalResponseFilter
- */
-public class QuarkusRequestFilter implements Handler<RoutingContext>, TransactionalSessionHandler {
-
-    private final Logger logger = Logger.getLogger(QuarkusRequestFilter.class);
-
-    private final Predicate<RoutingContext> contextFilter;
-
-    public QuarkusRequestFilter() {
-        this(null);
-    }
-
-    public QuarkusRequestFilter(Predicate<RoutingContext> contextFilter) {
-        this.contextFilter = contextFilter;
-    }
-
-    private final LongAdder rejectedRequests = new LongAdder();
-    private volatile boolean loadSheddingActive;
-
-    @Override
-    public void handle(RoutingContext context) {
-        if (ignoreContext(context)) {
-            context.next();
-            return;
-        }
-        // our code should always be run as blocking until we don't provide a better support for running non-blocking code
-        // in the event loop
-        try {
-            // When running in Quarkus dev mode, this will run on Vert.x default worker pool.
-            // When running in Quarkus production mode, this will run on the Quarkus executor pool.
-            // It should not call the Quarkus executor directly, as this prevents metrics for `worker_pool_*` to be collected.
-            // See https://github.com/quarkusio/quarkus/issues/34998 for a discussion.
-            context.vertx().executeBlocking(ctx -> {
-                try {
-                    runBlockingCode(context);
-                    ctx.complete();
-                } catch (Throwable ex) {
-                    ctx.fail(ex);
-                }
-            }, false, null);
-            if (loadSheddingActive) {
```

Comment:
I believe this PR broke the load shedding. See also #21962. We should've had probably tests for it.

CC @ahus1

Reply:
Load shedding is not officially support as of now. If this PR breaks it, just continue with this PR and we'll fix it (and officially support it) afterwards.